fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/PartitionPruneExpressionExtractor.java

```diff
@@ -0,0 +1,204 @@
+        if (newLeft == BooleanLiteral.TRUE || leftContext.containsUnEvaluableExpression) {
+            return rightContext.containsUnEvaluableExpression ? BooleanLiteral.TRUE : newRight;
+        }
+
+        // If right node contains non-partition slot or is TURE, just discard it.
+        if (newRight == BooleanLiteral.TRUE || rightContext.containsUnEvaluableExpression) {
+            return newLeft;
+        }
+
+        // both does not contains non-partition slot.
+        return new And(newLeft, newRight);
+    }
+
+    @Override
+    public Expression visitOr(Or node, Context parentContext) {
+        // handle left node
+        Context leftContext = new Context();
+        Expression newLeft = node.left().accept(this, leftContext);
+        // handle right node
+        Context rightContext = new Context();
+        Expression newRight = node.right().accept(this, rightContext);
+
+        // if anyone of them is TRUE or contains non-partition slot, just return TRUE.
+        if (newLeft == BooleanLiteral.TRUE || newRight == BooleanLiteral.TRUE
+                || leftContext.containsUnEvaluableExpression || rightContext.containsUnEvaluableExpression) {
+            return BooleanLiteral.TRUE;
+        }
+
+        return new Or(newLeft, newRight);
+    }
+
+    /**
+     * Context
+     */
+    @VisibleForTesting
+    public static class Context {
+        private boolean containsUnEvaluableExpression;
+    }
+
+    /**
+     * The detector only indicate that whether a predicate contains interested slots or not,
+     * and do not change the predicate.
+     */
+    @VisibleForTesting
+    public static class ExpressionEvaluableDetector extends DefaultExpressionRewriter<Context> {
+        private final Set<Slot> partitionSlots;
+
+        public ExpressionEvaluableDetector(Set<Slot> partitionSlots) {
+            this.partitionSlots = Objects.requireNonNull(partitionSlots);
+        }
+
+        /**
+         * Return true if expression does NOT contains un-evaluable expression.
+         */
+        @VisibleForTesting
+        public boolean detect(Expression expression) {
+            Context context = new Context();
+            expression.accept(this, context);
+            return !context.containsUnEvaluableExpression;
+        }
+
+        @Override
+        public Expression visit(Expression originExpr, Context parentContext) {
+            if (parentContext.containsUnEvaluableExpression) {
+                return originExpr;
+            }
+
+            // todo: Should we consider un-deterministic functions as un-evaluable expression?
+            // if (originExpr instanceof Function) {
+            //     Function function = (Function) originExpr;
+            //     if (Env.getCurrentEnv().isNondeterministicFunction(function.getName())) {
+            //         parentContext.containsUnEvaluableExpression = true;
+            //         return originExpr;
+            //     }
+            // }
```

Comment:
not need, some Nondeterministic functions are foldable and others are not. FoldConstantRuleOnFE can detect it.